wordpress-mobile/WordPress-Android
# [Reader Improvements] Fix Reader placeholder for recommended blog item and site header

Even after our implementation of Reader Improvements there were still two places showing the previous placeholder: site header and "You might like" sites. On site header the correct placeholder was being set on XML but before the image was loaded the previous placeholder was quickly shown before the image was loaded. This happened because the previous placeholder was being set programmatically.

### Before
#### Site header
https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/e7116205-8519-4cd5-86f4-563a3809a39a

#### You might like
<img width="347" alt="Screenshot 2023-10-31 at 5 28 28 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/0f974737-00e7-48e8-b1b5-333f1f613e23">


### After
#### Site header
https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/0b6923d6-6600-4b29-9246-f96d86b6ac5a

#### You might like
<img width="344" alt="Screenshot 2023-10-31 at 5 23 34 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/03cf60c6-b3db-445b-9992-f4d6709f9947">


To test:
1 - Install JP and sign in;
2 - Open reader -> "Discover";
3 - Scroll down to "You might like" sites section: you should see the correct placeholder;
4 - Open any site header by tapping on the site name: you should see the correct placeholder and the previous placeholder shouldn't appear at any point.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


UI Changes testing checklist:
